code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -167,6 +167,47 @@
 	R.updatehealth()
 	return
 
+/////////////////////////////////////////////
+// Agent Orange
+/////////////////////////////////////////////
+/obj/effect/particle_effect/smoke/agentorange
+	name = "agent orange smoke"
+	color = "#ff3b00"
+	time_to_live = 60 //hopefully this is about a minute
+/obj/effect/particle_effect/smoke/agentorange/Move()
+	..()
+	for(var/mob/living/carbon/human/R in get_turf(src))
+		affect(R)
+
+/obj/effect/particle_effect/smoke/agentorange/affect(var/mob/living/carbon/R)
+	..()
+	var/antispam
+	if(isXeno(R))
+		R.adjustFireLoss(R.maxHealth*(rand(0.02, 0.1))) // Eats through xeno health like candy because they're made of resin or it has a reaction with plasma or some peusdoscience bullshit
+	else if(ishuman(R))
+		var/mob/living/carbon/human/H = R
+		if(istype(H.wear_mask, /obj/item/clothing/mask/gas))
```

Comment:
not really the best way to do this but Im unsure if CM has some form of bitflag for specific protection types such as breathing filtering or skin protection